### PR TITLE
Enable backward compatibility for checkpoints with CompressionLevel

### PR DESCRIPTION
**Pull request by vanyalzr**

### CODE REVIEW

File: nncf/torch/compression_method_api.py [hunk 1]
```diff
@@ -112,15 +113,28 @@ def load_state(self, state: Dict[str, object]) -> None:
...
+            self.scheduler.load_state(state['scheduler'])
+        if 'compression_level' in state:
+            if 'compression_stage' not in state:
+                compression_level = state['compression_level']
+                state['compression_stage'] = CompressionLevel.map_legacy_level_to_stage()[compression_level]
+            else:
+                nncf_logger.warning('Both CompressionStage and (legacy) CompressionLevel attributes '
+                                    'are specified in the checkpoint. Proceeding with the value stored '
+                                    'in CompressionStage')
+        if self.compression_stage() != state['compression_stage']:
```
Comment (vanyalzr) [inline nncf/torch/compression_method_api.py:126]: @ljaljushkin Just adding a check of whether `CompressionStage` has the same value in the controller and in the checkpoint. This logic can probably be extended when the full controller state saving/loading functionality will be implemented.

**Review by ljaljushkin — APPROVED** [review]

File: nncf/api/compression.py [hunk 2]
```diff
@@ -329,3 +329,22 @@ def get_transformation_layout(self, model: ModelType) -> TransformationLayout:
         :return: The instance of the `TransformationLayout` class containing
             a list of algorithm-specific modifications.
         """
+
+
+class CompressionLevel(OrderedEnum):
```
Comment (alexsu52) [inline nncf/api/compression.py:334]: Could you add a description that the class is deprecated and will be removed in the future release? Please, open the ticket on this task if we have not yet.

**Review by alexsu52 — APPROVED** [review]

**Comment by vanyalzr** [comment]
@vshampor ready for merge

**Comment by vanyalzr** [comment]
@vshampor ready for merge

**Review by vshampor — APPROVED** [review]

File: nncf/torch/composite_compression.py [hunk 3]
```diff
@@ -144,3 +144,8 @@ def apply_to(self, target_model: NNCFNetwork) -> NNCFNetwork:
         for ctrl in self.child_ctrls:
             target_model = ctrl.apply_to(target_model)
         return target_model
+
+    def load_state(self, states):
+        self._check_loaded_compression_stage(states)
+        for child_ctrl, child_state in zip(self.child_ctrls, states['scheduler']):
+            child_ctrl.load_state({'scheduler': child_state})
```
Comment (andrey-churkin) [inline nncf/torch/composite_compression.py:151]: @vanyalzr Have you updated the checkpoints according to these changes?

Comment (vanyalzr) [reply nncf/torch/composite_compression.py:151]: PyTorch checkpoints with the composite compression algorithm already had the `scheduler` field as a list, it's just that this list is now in `states['scheduler']`, not in just `states`, hence no need to update checkpoints here

**Pull request merged**